### PR TITLE
feat: add meson-python backend

### PR DESCRIPTION
**Pull request by henryiii**
Adding meson-python backend.

### CODE REVIEW

**Comment by henryiii** [comment]
On windows: 

```
c++: fatal error: '-fuse-linker-plugin', but liblto_plugin.dll not found
```

File: {{cookiecutter.project_name}}/meson-mesonpy.build [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+py_mod = import('python')
+py = py_mod.find_installation(pure: false)
+
+pybind11_config = find_program('pybind11-config')
+pybind11_config_ret = run_command(pybind11_config, ['--includes'], check: true)
+pybind11 = declare_dependency(
+    include_directories: [pybind11_config_ret.stdout().split('-I')[-1].strip()],
+)
+py.install_sources(
```
Comment (eli-schwartz) [inline {{cookiecutter.project_name}}/meson-mesonpy.build:22]: There are two schools of thought when it comes to installing files. One is the "detect a source tree" approach (e.g. setuptools `find_packages()`), and one is the "explicitly list all files" approach.

This is the latter approach. The former approach can be accomplished in meson with:

```meson
install_subdir('src', install_dir: py.get_install_dir())
```

Without wearing my meson hat... I personally recommend listing all sources explicitly, guesswork is too often an issue especially when you e.g. forget to check new files into git, and then both your local tree and the version you pushed to a public software forge build successfully, but do different things.

**Comment by eli-schwartz** [comment]
> Unlike Setuptools and CMake, Meson can't find MSVC unless it's in a MSVC command prompt.

You have to opt in, really. `meson setup --vsenv builddir/`. This will force MSVC detection.

If you don't force it, meson will check for different compilers trying to figure out whether you want MSVC. If it sees GCC or clang installed it assumes you manually installed that and want to prefer that. Otherwise, it falls back to setting up the vsenv.

This assumption breaks hard in GitHub Actions, which provides 3 or 4 different GCC/clang compilers, including the infamously broken Strawberry Perl one.

File: {{cookiecutter.project_name}}/meson-mesonpy.build (hunk 1 above)
Comment (eli-schwartz) [reply {{cookiecutter.project_name}}/meson-mesonpy.build]: Yup, and meson is committed to supporting that use case, even if some of its developers wouldn't personally use it. :)

**Pull request merged**